### PR TITLE
Fix wc networking

### PR DESCRIPTION
**Pull request by ERussel**
- improve wallet connect reconnection handling #85ztr6mb0
- fix freezes on sign bytes when bio auth enabled #85ztqbdge
- add support for auth for ethereum sign bytes

### CODE REVIEW

**Comment by stepanLav** [comment]
Task linked: [CU-85ztr6mb0 iOS. Wallet Connect. Can not connect to dapp due to network issues](https://app.clickup.com/t/85ztr6mb0)

**Comment by stepanLav** [comment]
Task linked: [CU-85ztqbdge iOS. When active approve with PIN sign bytes in dapps freezes or appoves without PIN](https://app.clickup.com/t/85ztqbdge)

File: novawallet/Common/Services/WalletConnect/WalletConnectService.swift [hunk 1]
```diff
@@ -262,105 +262,169 @@ extension WalletConnectService: WalletConnectServiceProtocol {
...
                 self?.markDisconnectedAndNotify(error: nil)
+            case .reconnectSuggested:
+                self?.stopWebsocket()
+                self?.startWebsocket()
+            case let .viabilityChanged(isViable):
+                if isViable {
+                    self?.stopWebsocket()
+                    self?.startWebsocket()
+                } else {
+                    self?.markDisconnectedAndNotify(error: nil)
```
Comment (valentunn) [inline novawallet/Common/Services/WalletConnect/WalletConnectService.swift:356]: Should this be guarded by mutex too?

**Review by valentunn — APPROVED** [review]

File: novawallet/Common/Services/WalletConnect/WalletConnectService.swift (hunk 1 above)
Comment (ERussel) [reply novawallet/Common/Services/WalletConnect/WalletConnectService.swift]: Yes, missed this in the corner case

**Pull request merged**